groups/validate.go

```diff
@@ -104,196 +134,232 @@ func TestMergedGroupsConfig(t *testing.T) {
+			fmt.Print(err)
+			os.Exit(1)
 		}
 	}
 }
 
 // Enforce conventions for all groups
-func TestGroupConventions(t *testing.T) {
-	for _, g := range cfg.Groups {
+func VerifyGroupConventions(config Config) {
+	primaryDomain := config.Domains[0]
+	errs := 0
+	for _, g := range groupsConfig.Groups {
 		// groups are easier to reason about if email and name match
-		expectedEmailId := g.Name + "@kubernetes.io"
-		if g.EmailId != expectedEmailId {
-			t.Errorf("group '%s': expected email '%s', got '%s'", g.Name, expectedEmailId, g.EmailId)
+		expectedEmailID := g.Name + "@" + primaryDomain
+		if g.EmailId != expectedEmailID {
+			log.Printf("group '%s': expected email '%s', got '%s'", g.Name, expectedEmailID, g.EmailId)
+			errs++
 		}
 	}
+	if errs > 0 {
+		os.Exit(1)
+	}
 }
 
 // Enforce conventions for all k8s-infra groups
-func TestK8sInfraGroupConventions(t *testing.T) {
-	for _, g := range cfg.Groups {
+func VerifyK8sInfraGroupConventions(config Config) {
+	errs := 0
+	for _, g := range groupsConfig.Groups {
 		if strings.HasPrefix(g.EmailId, "k8s-infra") {
 			// no owners because we want to prevent manual membership changes
 			if len(g.Owners) > 0 {
-				t.Errorf("group '%s': must have no owners, only members", g.Name)
+				log.Printf("group '%s': must have no owners, only members", g.Name)
+				errs++
 			}
 
 			// treat files here as source of truth for membership
 			reconcileMembers, ok := g.Settings["ReconcileMembers"]
 			if !ok || reconcileMembers != "true" {
-				t.Errorf("group '%s': must have settings.ReconcileMembers = true", g.Name)
+				log.Printf("group '%s': must have settings.ReconcileMembers = true", g.Name)
+				errs++
 			}
 		}
 	}
+	if errs > 0 {
+		os.Exit(1)
+	}
 }
 
 // Enforce conventions for groups used by GKE Group-based RBAC
 // - there must be a gke-security-groups@ group
 // - its members must be k8s-infra-rbac-*@ groups (and vice-versa)
 // - all groups involved must have settings.WhoCanViewMembership = ALL_MEMBERS_CAN_VIEW
-func TestK8sInfraRBACGroupConventions(t *testing.T) {
+func VerifyK8sInfraRBACGroupConventions(config Config) {
+	primaryDomain := config.Domains[0]
 	rbacEmails := make(map[string]bool)
-	for _, g := range cfg.Groups {
+	errs := 0
+	for _, g := range groupsConfig.Groups {
 		if strings.HasPrefix(g.EmailId, "k8s-infra-rbac") {
 			rbacEmails[g.EmailId] = false
 			// this is necessary for group-based rbac to work
 			whoCanViewMembership, ok := g.Settings["WhoCanViewMembership"]
 			if !ok || whoCanViewMembership != "ALL_MEMBERS_CAN_VIEW" {
-				t.Errorf("group '%s': must have settings.WhoCanViewMembership = ALL_MEMBERS_CAN_VIEW", g.Name)
+				log.Printf("group '%s': must have settings.WhoCanViewMembership = ALL_MEMBERS_CAN_VIEW", g.Name)
+				errs++
 			}
 		}
 	}
 	foundGKEGroup := false
-	for _, g := range cfg.Groups {
-		if g.EmailId == "gke-security-groups@kubernetes.io" {
+	for _, g := range groupsConfig.Groups {
+		if g.EmailId == "gke-security-groups@"+primaryDomain {
```

Comment:
I'm not sure swapping this for `primaryDomain` helps much in terms of portability. I'd leave this as k8s-specific and just opt people out of this test.

Reply:
This is a useful for feature for anyone doing [Google Groups RBAC for GKE](https://cloud.google.com/kubernetes-engine/docs/how-to/google-groups-rbac). We'll be using this in Knative. In GSuite, there is a primary domain and secondary domains and key features are tied to the primary domain(item 0 in the list)

https://support.google.com/a/answer/7009324?hl=en